py/objtype.c

```diff
@@ -1011,6 +1011,31 @@ STATIC void type_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
             dest[0] = MP_OBJ_NEW_QSTR(self->name);
             return;
         }
+        if (attr == MP_QSTR___bases__) {
+            const void * parent = self->parent;
```

Comment:
You may be able to reduce the code size slightly, here's a rough idea:

```c
mp_obj_t parent_obj = self->parent ? MP_OBJ_FROM_PTR(self->parent) : MP_OBJ_FROM_PTR(&mp_type_object);
#if MICROPY_MULTIPLE_INHERITANCE
if (mp_obj_is_type(parent_obj, &mp_type_tuple)) {
    dest[0] = parent_obj;
} else {
#endif
    dest[0] = mp_obj_new_tuple(1, &parent_obj);
}
```

Reply:
I get a saving of 12 bytes on STM32. (Only lightly tested)

Reply:
I'm not sure if this reads better or not, but I am receiving your concern about code size. I'll see what I can do to minimize the footprint this feature imposes.